modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPathAttributeGenerator.cs

```diff
@@ -60,6 +60,10 @@ public void Execute(GeneratorExecutionContext context)
 
             foreach (var godotClass in godotClasses)
             {
+                // Ignore generic classes.
+                if (godotClass.Key.IsGenericType)
```

Comment:
This is problematic, right? Because now we don't have `[ScriptPathAttribute(...)]` for generic classes. What's problematic in the test is we have several classes defined in the same file (thus, several types with the same `[ScriptPathAttribute(...)]` value), not that the attribute is generated for generics.

Reply:
You are right. I was thinking we didn't need the attribute for the generic types because they use the virtual path, but we actually use it to generate that virtual path :upside_down_face:.

Reply:
Is this an issue that we can now generate several `[ScriptPathAttribute(...)]` pointing to the same path? If so, we should probably consider adding a diagnostic, failing source generation, and showing an understandable error 🤔

Reply:
Well, adding to `_pathTypeBiMap` with the same key will throw an exception. This would normally not be an issue because we already require your class name to match the name of the file and you can't have multiple classes with the same name.

Technically you can have multiple classes with the same name in the same file if they are under different namespace blocks but that's probably extremely rare.

However, now that we're generating the path attribute for generic types, this makes it possible to define multiple classes with the same name and I think this use case is a lot more common. So I agree we should do something about it.

Reply:
That can probably be done in a separate PR 👍

Reply:
- Opened https://github.com/godotengine/godot/pull/88371.